codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientDocs.kt

```diff
@@ -0,0 +1,224 @@
+            throttling, or other middleware-like tasks.
+            - A retry strategy (`R`) dictates the behavior for requests that fail. The default,
+            [`RetryMode::Standard`](aws_smithy_types::retry::RetryMode::Standard) is generally what you want since
+            it provides a well-vetted exponential backoff retry implementation.
+
+            Once the Smithy client is created, a service config and fluent client can be created. Generally, you
+            want to call [`Client::with_config`], which takes a Smithy client and the service [`Config`](crate::Config).
+            The config is constructed using the [builder pattern], and has several builder methods to further
+            customize the client.
+
+            In _most_ circumstances, you will want to use the following pattern to construct a client:
+
+            ```rust,no_run
+            let smithy_client = $moduleUseName::client::Builder::new()
+                .dyn_https_connector(Default::default())
+            ##  /*
+                .middleware(/* discussed below */)
+            ##  */
+            ##  .middleware_fn(|r| r)
+                .build_dyn();
+
+            let config = $moduleUseName::Config::builder().build();
+            let client = $moduleUseName::Client::with_config(smithy_client, config);
+            ```
+
+            _Note:_ Client construction is expensive due to connection thread pool initialization, and should be done
+            once at application start-up.
+
+            For middleware, you'll want to use whatever matches the routing, authentication, and authorization
+            required by the target service. For example, for the AWS SDK which uses [SigV4-signed requests], the
+            middleware looks like this:
+
+            ```rust,ignore
+            use aws_endpoint::AwsEndpointStage;
+            use aws_http::auth::CredentialsStage;
+            use aws_http::recursion_detection::RecursionDetectionStage;
+            use aws_http::user_agent::UserAgentStage;
+            use aws_sig_auth::middleware::SigV4SigningStage;
+            use aws_sig_auth::signer::SigV4Signer;
+            use aws_smithy_client::retry::Config as RetryConfig;
+            use aws_smithy_http_tower::map_request::{AsyncMapRequestLayer, MapRequestLayer};
+            use std::fmt::Debug;
+            use tower::layer::util::{Identity, Stack};
+            use tower::ServiceBuilder;
+
+            type AwsMiddlewareStack = Stack<
+                MapRequestLayer<RecursionDetectionStage>,
+                Stack<
+                    MapRequestLayer<SigV4SigningStage>,
+                    Stack<
+                        AsyncMapRequestLayer<CredentialsStage>,
+                        Stack<
+                            MapRequestLayer<UserAgentStage>,
+                            Stack<MapRequestLayer<AwsEndpointStage>, Identity>,
+                        >,
+                    >,
+                >,
+            >;
```

Comment:
@drganjoo Do y'all have a recommended starter middleware stack for generic clients?